src/coreclr/jit/llvm.cpp

```diff
@@ -759,62 +806,156 @@ Value* buildCall(llvm::IRBuilder<>& builder, GenTree* node)
             // replacement for _info.compCompHnd->recordRelocation(nullptr, gtCall->gtEntryPoint.handle, IMAGE_REL_BASED_REL32);
             (*_addCodeReloc)(_thisPtr, call->gtEntryPoint.handle);
 
-            return mapGenTreeToValue(node, builder.CreateCall(llvmFunc, getShadowStackForCallee(builder)));
+            mapGenTreeToValue(node, builder.CreateCall(llvmFunc, getShadowStackForCallee(builder)));
+            return;
         }
     }
     else if (call->gtCallType == CT_USER_FUNC && !call->IsVirtualStub() /* TODO: Virtual stub not implemented */)
     {
-        return buildUserFuncCall(call, builder);
+        buildUserFuncCall(call, builder);
+        return;
     }
     failFunctionCompilation();
 }
 
-Value* buildCnsInt(llvm::IRBuilder<>& builder, GenTree* node)
+void buildCast(llvm::IRBuilder<>& builder, GenTreeCast* cast)
+{
+    if (cast->gtCastType == TYP_BOOL && cast->TypeGet() == TYP_INT && cast->gtGetOp1()->TypeGet() == TYP_INT)
+    {
+        Value* intValue = builder.CreateZExt(getGenTreeValue(cast->gtGetOp1()).getValue(builder), getLLVMTypeForVarType(TYP_INT));
+        mapGenTreeToValue(cast, intValue); // nothing to do except map the source value to the destination GenTree
+    }
+    else
+    {
+        // TODO: other casts
+        failFunctionCompilation();
+    }
+}
+
+void buildCnsInt(llvm::IRBuilder<>& builder, GenTree* node)
 {
     if (node->gtType == var_types::TYP_INT)
     {
-        return mapGenTreeToValue(node, builder.getInt32(node->AsIntCon()->gtIconVal));
+        mapGenTreeToValue(node, builder.getInt32(node->AsIntCon()->gtIconVal));
+        return;
```

Comment:
A bit more portable way would be to add a `static_cast<int32_t>`, since `gtIconVal` is a `nint` in C# terms, but it works as is and will error out when porting to 64 bit, so, eh.
```suggestion
        mapGenTreeToValue(node, builder.getInt32(node->AsIntCon()->IconValue()));
        return;
```